src/core/src/op/assign.cpp

```diff
@@ -73,18 +73,16 @@ void Assign::validate_and_infer_types() {
     const auto& input_type = get_input_element_type(0);
     const auto& input_shape = get_input_partial_shape(0);
 
-    OPENVINO_ASSERT(input_type.compatible(variable_type),
-                    "The type specified in the Variable is not compatible with the input type.",
-                    " Input type: ",
-                    input_type,
-                    " Variable type: ",
-                    variable_type);
-    OPENVINO_ASSERT(input_shape.compatible(variable_shape),
-                    "The shape specified in the Variable is not compatible with the shape of the input.",
-                    " Input shape: ",
-                    input_shape,
-                    " Variable shape: ",
-                    variable_shape);
+    NODE_VALIDATION_CHECK(this,
+                          input_type.compatible(variable_type),
+                          "The type specified in the Variable is not compatible with the input type."
+                          " Variable type: ",
```

Comment:
it's better to print information about the input shape/type, can be useful for debugging purposes

Reply:
Yes, it will print information about inputs (shapes, types) unless default `ctor` use then will be empty information as there is no inputs.
example:
`While validating node 'opset6::Assign Assign_1213634 (ReadValue_1213633[0]:f16[4,3,2,1]) -> (dynamic[...])' with friendly_name 'Assign_1213634':`

after this the user message is added in new line.